tests/unit/test_cli.py

```diff
@@ -0,0 +1,263 @@
+import pytest
+import sys
+import os
+from unittest.mock import patch, MagicMock, mock_open
+
+# Add the src directory to the sys.path
+sys.path.insert(
+    0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../src"))
+)
+
+from ublue_update.cli import (
+    notify,
+    ask_for_updates,
+    inhibitor_checks_failed,
+    run_updates,
+)
+
+
+@patch("ublue_update.cli.cfg")
+@patch("ublue_update.cli.os")
+@patch("ublue_update.cli.log")
+@patch("ublue_update.cli.subprocess.run")
+def test_notify_no_dbus_notify(mock_run, mock_log, mock_os, mock_cfg):
+    mock_cfg.dbus_notify = False
+    assert notify("test_title", "test_body") == None
+
+
+@patch("ublue_update.cli.cfg")
+@patch("ublue_update.cli.os")
+@patch("ublue_update.cli.log")
+@patch("ublue_update.cli.subprocess.run")
+def test_notify_uid_user(mock_run, mock_log, mock_os, mock_cfg):
+    title = "test title"
+    body = "test body"
+    mock_cfg.dbus_notify = True
+    mock_os.getuid.return_value = 1001
+    notify(title, body)
+    mock_run.assert_called_once_with(
```

Comment:
I like the idea of checking to see if the command has been ran properly, but this could also be a problem down the line if someone changes the arguments or if notify-send changes its cli. This also goes for a lot of these other subprocess calls. Testing subprocess is hard though and likely will need to be solved by integration tests down the line

Reply:
ya its definitely tricky and the trap with unit tests at times is asserting too much as it makes the tests more inflexible and requires more maintenance on them.  I could reduce the number of assertions to a more simple input -> output format that would allow changes to not break the tests as much.